src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1996,36 +1996,53 @@ namespace winrt::TerminalApp::implementation
         _HookupKeyBindings(_settings.ActionMap());
 
         // Refresh UI elements
-        auto profiles = _settings.ActiveProfiles();
-        for (const auto& profile : profiles)
+
+        // Mapping by GUID isn't _excellent_ because the defaults profile doesn't have a stable GUID; however,
+        // when we stabilize its guid this will become fully safe.
+        std::unordered_map<winrt::guid, std::pair<Profile, TerminalSettingsCreateResult>> _profileGuidSettingsMap;
```

Comment:
Remove the underscore here - it's a local variable right?

_If_ you want to you can do:
```cpp
const auto profileDefaults{ _settings.ProfileDefaults() };
const auto allProfiles{ _settings.AllProfiles() };

std::unordered_map<winrt::guid, std::pair<Profile, TerminalSettingsCreateResult>> profileGuidSettingsMap;
profileGuidSettingsMap.reserve(allProfiles.Size() + 1); // <------------------

// Abstract away insert_or_assign? Optional ¯\_(ツ)_/¯
static constexpr auto add = [](auto& map, const auto& profile) {
	map.insert_or_assign(profile.Guid(), std::pair{ profile, nullptr });
};

add(profileGuidSettingsMap, profileDefaults);
for (const auto& newProfile : allProfiles)
{
    // Avoid creating a TerminalSettings right now. They're not totally cheap, and we suspect that users with many
    // panes may not be using all of their profiles at the same time. Lazy evaluation is king!
    add(profileGuidSettingsMap, newProfile);
}
```

Reply:
Underscore: huh, I have no idea why I did that.

I'm happy to do the reservation, but the abstracting-away of `insert_or_assign` trades +4 lines of code and a bunch of automatic deduction for clearing up two other lines of code. I'm iffy on that one. :smile:

Thanks!